sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -639,10 +648,21 @@ private HttpPipeline setupPipeline(HttpClient httpClient) {
      * @param port the port on which the HTTP server is listening
      * @return a Publisher that emits an AccessToken
      */
-    public Mono<MsalToken> authenticateWithBrowserInteraction(TokenRequestContext request, int port) {
+    public Mono<MsalToken> authenticateWithBrowserInteraction(TokenRequestContext request, Integer port,
+                                                              String redirectURL) {
         URI redirectUri;
+        String redirect;
+
+        if (port != null) {
+            redirect = HTTP_LOCALHOST + ":" + port;
+        } else if (redirectURL != null) {
+            redirect = redirectURL;
+        } else {
+            redirect = HTTP_LOCALHOST;
```

Comment:
I wouldn't default to `localhost` here. We are secretly setting a default here and it will throw an error from AAD saying the redirect URL is not valid. The user will probably scratch the head wondering where he has set the redirect URL. A better error message would be "Either a port or a redirect URl must be provided on `InteractiveBrowserCredentialBuilder`, which leads them to read the javadocs there and understand what they need to do. 

You could also do this validation in the `build()` method on `InteractiveBrowserCredential`.

Reply:
See https://github.com/Azure/azure-sdk-for-java/pull/14657/files#r482323637. 

The bottom line is - there is no default value for redirectUrl or port - because it can be any arbitrary value the user specifies on the app registration.

Reply:
So, the redirect URL is only required, if user is trying to authenticate with their custom registered application.
If they are using a first party client id, then the redirectURL can be arbitrary in the sense that Localhost can be used with any port and user doesn't need to supply it.

Javadocs provides this guidance too, only for custom client ids, redirect URL is required to be supplied.